cartridge/stateboard-client.lua

```diff
@@ -90,6 +90,69 @@ local function get_coordinator(session)
     return coodinator
 end
 
+local function set_vclockkeeper(session, replicaset_uuid, instance_uuid, vclock)
+    checks('stateboard_session', 'string', 'string', '?table')
+    assert(session.connection ~= nil)
+
+    local keeper, err = errors.netbox_call(session.connection,
```

Comment:
I don't understand a bit this part. Why do you call "get" and then "set" (two network calls). I assume you could do it via single call. Even it will be called "compare_and_set".

Reply:
Well, I could move it outside the client, but it would save us only one `get_vclockkeeper` call in `failover.lua`. The coordinator role still has to query keepers before setting them because it _must_ be protected with CaS. We don't want any late TCP packet to void consistency.

I can try to optimize it later, but it would just increase quantity of code.

Reply:
> We don't want any late TCP packet to void consistency.

If I correctly understand you it could be eliminated via introducing critical section here.
Anyway if you want to stay it as is please file a follow-up issue to don't forget to optimize it in future.